Fix support for bignum array

sydhds opened the pull request.

peterjah reviewed (changes requested):
a lot of code duplication.
I think it would have been possible to keep a generic templated function.
the byte buffer size of a class car be retrieved with `offsetof<T>()` 

You keeped `nextFixedSizeArray` but we now its not working with some types, maybe adding a check on types would be great to avoid futur bugs.

it would be also nice to complete/add unit tests that cover at least the bug you are fixing,

sydhds commented:
> a lot of code duplication. I think it would have been possible to keep a generic templated function. the byte buffer size of a class can be retrieved with `offsetof<T>()`
> 
> You keeped `nextFixedSizeArray` but we know its not working with some types, maybe adding a check on types would be great to avoid future bugs.
> 
> it would be also nice to complete/add unit tests that cover at least the bug you are fixing,

I've tried the generic function but the compiler rejects it. I've updated the code to use offsetof() and updated the unit tests as well.

sydhds reviewed (commented):
I would remove support for i256 as it is almost unimplemented in as-bignum...

peterjah commented on packages/as-types/assembly/__tests__/argument.spec.ts:
```diff
@@ -496,23 +504,4 @@ describe('Args tests', () => {
     expect(deser[1].name).toBe('Superman');
   });
-
-  it('With array of bytes', () => {
```
Yes i removed the support of serialize array of fixed size StaticArray... i don't see any usecase to this.
And its may be source of bug if the user provide elements of different sizes...

peterjah commented on packages/as-types/assembly/serialization/bignum.ts:
```diff
@@ -12,6 +13,17 @@ export function bytesToU128(arr: StaticArray<u8>): u128 {
+ * @returns the converted i128
+ */
+export function bytesToI128(arr: StaticArray<u8>): i128 {
+  return changetype<i128>(u128.from(wrapStaticArray(arr)));
```
i128 use same serialization of u128. its just the way we interprets the bytes that change
